src/frontend/src/handler/create_connection.rs

```diff
@@ -0,0 +1,111 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+
+use pgwire::pg_response::{PgResponse, StatementType};
+use risingwave_common::error::ErrorCode::ProtocolError;
+use risingwave_common::error::{Result, RwError};
+use risingwave_connector::source::kafka::PRIVATELINK_CONNECTION;
+use risingwave_pb::ddl_service::create_connection_request;
+use risingwave_sqlparser::ast::CreateConnectionStatement;
+use serde_json;
+
+use super::RwPgResponse;
+use crate::binder::Binder;
+use crate::handler::HandlerArgs;
+
+pub(crate) const CONNECTION_TYPE_PROP: &str = "type";
+pub(crate) const CONNECTION_PROVIDER_PROP: &str = "provider";
+pub(crate) const CONNECTION_SERVICE_NAME_PROP: &str = "service.name";
+pub(crate) const CONNECTION_AVAIL_ZONE_PROP: &str = "availability.zones";
+
+#[inline(always)]
+fn get_connection_property_required(
+    with_properties: &HashMap<String, String>,
+    property: &str,
+) -> Result<String> {
+    with_properties
+        .get(property)
+        .map(|s| s.to_lowercase())
+        .ok_or(RwError::from(ProtocolError(format!(
+            "Required property \"{property}\" was not provided"
+        ))))
+}
+
+fn resolve_private_link_properties(
+    with_properties: &HashMap<String, String>,
+) -> Result<create_connection_request::PrivateLink> {
+    let provider = get_connection_property_required(with_properties, CONNECTION_PROVIDER_PROP)?;
+    let service_name =
+        get_connection_property_required(with_properties, CONNECTION_SERVICE_NAME_PROP)?;
+    let availability_zones_str =
+        get_connection_property_required(with_properties, CONNECTION_AVAIL_ZONE_PROP)?;
+    let availability_zones: Vec<String> =
+        serde_json::from_str(&availability_zones_str).map_err(|e| {
+            RwError::from(ProtocolError(format!(
+                "Can not parse {}: {}",
+                CONNECTION_AVAIL_ZONE_PROP, e
+            )))
+        })?;
```

Comment:
should we check with aws to see if the az is misspelled?

Reply:
I guess we don't have to check it in the frontend, since if the az is mispelled, the connection created will not contain the mispelled az. The user can check the result using `show connection` afterwards :)

Reply:
kafka brokers may distribute in multiple AZs. If we cannot access one of the brokers, it can lead to failures.
If so, we should check accessibility for each broker when creating a source. Is it acceptable? cc @StrikeW

Reply:
I find that we can list the available AZs for the endpoint service, then user doesn't need to specify the `availability.zones` in SQL. This can be enhanced in the following PR.